pg_search/src/postgres/storage/buffer.rs

```diff
@@ -692,7 +689,9 @@ impl BufferManager {
                 // by extending the relation with brand new buffers
                 new_buffers = Some(bman.buffer_access().new_buffers(remaining_from_fsm).map(
                     move |pg_buffer| {
-                        block_tracker::track!(Write, pg_buffer);
+                        block_tracker::track!(Write, unsafe {
+                            pg_sys::BufferGetBlockNumber(pg_buffer)
+                        });
```

Comment:
Not something to try and do here probably, but it feels like the whole buffer tracking system should potentially be in the `RelationBufferAccess` module instead: it has slightly less surface area, and all of the methods here should probably be using the abstractions that it provides instead.